### PR TITLE
Support Stardew Valley 1.6 / SMAPI 4.0

### PR DESCRIPTION
**Pull request by aschismatism**
This also includes changes from #16 , which was the last unofficial release.

### CODE REVIEW

File: ConvenientChests/CraftFromChests/MenuListener.cs [hunk 1]
```diff
@@ -38,32 +44,25 @@ public class MenuListener {
...
 
             switch (e.NewMenu) {
                 case GameMenu _:
-                    GameMenuShown?.Invoke(sender, e);
-                    this.Events.Display.RenderedActiveMenu += OnRenderedActiveMenu;
+                    Events.Display.RenderedActiveMenu += OnRenderedActiveMenu;
                     break;
 
-                case CraftingPage _:
                 case object m when m.GetType().ToString() == "CookingSkill.NewCraftingPage":
```
Comment (aEnigmatic) [inline ConvenientChests/CraftFromChests/MenuListener.cs:56]: Probably unnecessary now with the `p.cooking` field, but I couldn't remember if this was from the base game or some other cooking mod

File: ConvenientChests/CategorizeChests/Interface/WidgetHost.cs [hunk 2]
```diff
@@ -5,28 +5,29 @@
...
+    class WidgetHost : BaseOverlay {
         public readonly Widget RootWidget;
         public readonly ITooltipManager TooltipManager;
 
-        public WidgetHost(IModEvents events, IInputHelper input)
-            : base(events, input)
-        {
-            RootWidget = new Widget() {Width = Game1.viewport.Width, Height = Game1.viewport.Height};
+        public WidgetHost(IModEvents events, IInputHelper input, IReflectionHelper reflection)
+            : base(events, input, reflection, assumeUiMode: true) {
```
Comment (aEnigmatic) [inline ConvenientChests/CategorizeChests/Interface/WidgetHost.cs:14]: `assumeUiMode: true` fixes the mouse click coordinates.  
The rest is just updating the BaseOverlay class to the newest version (thanks again @Pathoschild 👍) and me commiting my codestyles 👎

**Comment by aEnigmatic** [comment]
Thanks a lot for the pull request 👍 
If you have the time, I'd appreciate some testing, as I don't have a real savegame and no mods installed :)

**Comment by aschismatism** [comment]
Thanks for taking a quick look! I'll be doing a bit more thorough testing; I didn't really expect a reply.

~I know that zoom levels are broken right now, but the general base functionality seems okay. I'll tag you once I've done more thorough testing; in the meanwhile, I'm tracking feedback I get from [the NexusMods posts](https://www.nexusmods.com/stardewvalley/mods/2196?tab=posts) and [in the forum thread](https://forums.stardewvalley.net/threads/unofficial-mod-updates.2096/page-129#post-120598).~

Somehow didn't notice you added commits. Thanks for the explanation on `assumeUiMode` :)

It's looking good to me; just pushed a small fix to allow tooltips to render again.

**Pull request merged**